test/object-store/util/sync/redirect_server.hpp

```diff
@@ -231,31 +314,35 @@ class RedirectingHttpServer {
             if (ec == util::error::operation_aborted) {
                 return;
             }
+            // Allow additonal connections to be accepted
             do_accept();
             if (ec) {
-                m_logger->error("Error accepting new connection in: %1", ec);
+                m_logger->error("Error accepting new connection to %1 [%2]: %3", base_url(), ec, ec.message());
                 return;
             }
 
             conn->http_server.async_receive_request([this, conn](HTTPRequest req, std::error_code ec) {
+                static bool use_301 = false; // send 301 on first redirect response
```

Comment:
Why is this a static bool? Why do we only send a 301 on the first redirect response? Seems like there's a bunch of state added to this type but this is a global static?